x/interchainquery/keeper/msg_submit_query_response_test.go

```diff
@@ -0,0 +1,165 @@
+		goCtx: goCtx,
+		query: query,
+	}
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_Successful() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+
+	resp, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	s.Require().NoError(err)
+	s.Require().NotNil(resp)
+	s.Require().Equal(&types.MsgSubmitQueryResponseResponse{}, resp)
+
+	// check that the query is not in the store anymore, as it should be deleted
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx(), tc.query.Id)
+	s.Require().False(found)
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_UnknownId() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	tc.query.Id += "INVALID_SUFFIX" // create an invalid query id
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+
+	resp, err := s.GetMsgServer().SubmitQueryResponse(tc.goCtx, &tc.validMsg)
+	s.Require().NoError(err)
+	s.Require().NotNil(resp)
+	s.Require().Equal(&types.MsgSubmitQueryResponseResponse{}, resp)
+
+	// check that the query is STILL in the store, as it should NOT be deleted because the query was not found
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx(), tc.query.Id)
+	s.Require().True(found)
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_ExceededTtl() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	tc.query.Ttl = uint64(1) // set ttl to be expired
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+	exceeded, err := s.App.InterchainqueryKeeper.HasQueryExceededTtl(s.Ctx(), &tc.validMsg, tc.query)
+	s.Require().NoError(err)
+	s.Require().True(exceeded)
+
+	// check that the query is not in the store anymore, as it should be deleted
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx(), tc.query.Id)
+	s.Require().False(found)
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_NotExceededTtl() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	tc.query.Ttl = uint64(2545450064) * uint64(1000000000) // set ttl to August 2050, mult by nano conversion factor
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+	exceeded, err := s.App.InterchainqueryKeeper.HasQueryExceededTtl(s.Ctx(), &tc.validMsg, tc.query)
+	s.Require().NoError(err)
+	s.Require().False(exceeded)
+
+	// check that the query is not in the store anymore, as it should be deleted
+	_, found := s.App.InterchainqueryKeeper.GetQuery(s.Ctx(), tc.query.Id)
+	s.Require().True(found)
+}
+
+func (s *KeeperTestSuite) TestMsgSubmitQueryResponse_FindAndInvokeCallback_Success() {
+	tc := s.SetupMsgSubmitQueryResponse()
+
+	s.App.InterchainqueryKeeper.SetQuery(s.Ctx(), tc.query)
+
+	// rather than testing by executing the callback in its entirety,
+	//   check by invoking it without a registered host zone and catching the appropriate error
+	err := s.App.InterchainqueryKeeper.FindAndInvokeCallback(s.Ctx(), &tc.validMsg, tc.query)
+	s.Require().ErrorContains(err, "WithdrawalBalanceCallback: no registered zone for chain id", "callback was invoked")
```

Comment:
Where does the host zone come into play here (i.e. what about this test setup is different than the others)? Also I think the function name might be wrong